client/src/main/java/org/apache/rocketmq/client/consumer/DefaultMQPushConsumer.java

```diff
@@ -417,7 +417,7 @@ public void createTopic(String key, String newTopic, int queueNum) throws MQClie
     @Override
     public void setUseTLS(boolean useTLS) {
         super.setUseTLS(useTLS);
-        if (traceDispatcher != null && traceDispatcher instanceof AsyncTraceDispatcher) {
```

Comment:
I thanks 'traceDispatcher != null' is still needed

Reply:
IMO,Null is not an instanceof anything, so a null check is redundant.

Reply:
> IMO,Null is not an instanceof anything, so a null check is redundant.

Agree with